examples/NAS-Envelopenet-trainingcontainer/model_constructor.py

```diff
@@ -0,0 +1,238 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from datetime import datetime
+import ast
+import sys
+import time
+import json
+
+import tensorflow as tf
+import net
+from evaluate import Evaluate
+global_batch_size = 128
+global_log_frequency = 10
+
+class _LoggerHook(tf.train.SessionRunHook):
+    """Logs loss and runtime."""
+
+    def __init__(self, global_step_init, loss):
+        self.global_step_init = global_step_init
+        self.loss = loss
+
+    def begin(self):
+        self._step = self.global_step_init
+        self._start_time = time.time()
+
+    def before_run(self, run_context):
+        self._step += 1
+        # Asks for loss value.
+        return tf.train.SessionRunArgs(self.loss)
+
+    def after_run(self, run_context, run_values):
```

Comment:
function is never called

Reply:
these are required in loggerhook call during training